src/webui/api/torrentscontroller.cpp

```diff
@@ -835,3 +856,91 @@ void TorrentsController::removeCategoriesAction()
     for (const QString &category : categories)
         BitTorrent::Session::instance()->removeCategory(category);
 }
+
+QVariantList TorrentsController::getStickyTrackers(const BitTorrent::TorrentHandle* torrent)
+{
+    static const QString working("Working");
+    static const QString disabled("Disabled");
+    static const QString privateMsg("This torrent is private");
+
+    QVariantList stickyTrackers;
+
+    QVariantMap dht;
+    dht[KEY_TRACKER_URL] = "** [DHT] **";
+    dht[KEY_TRACKER_TIER] = "";
+    dht[KEY_TRACKER_RECEIVED] = "";
+    dht[KEY_TRACKER_MSG] = "";
+    dht[KEY_TRACKER_DOWNLOADED] = "";
+
+    QVariantMap pex;
+    pex[KEY_TRACKER_URL] = "** [PeX] **";
+    pex[KEY_TRACKER_TIER] = "";
+    pex[KEY_TRACKER_RECEIVED] = "";
+    pex[KEY_TRACKER_MSG] = "";
+    pex[KEY_TRACKER_DOWNLOADED] = "";
+
+    QVariantMap lsd;
+    lsd[KEY_TRACKER_URL] = "** [LSD] **";
+    lsd[KEY_TRACKER_TIER] = "";
+    lsd[KEY_TRACKER_RECEIVED] = "";
+    lsd[KEY_TRACKER_MSG] = "";
+    lsd[KEY_TRACKER_DOWNLOADED] = "";
+
+    if (BitTorrent::Session::instance()->isDHTEnabled() && !torrent->isPrivate())
+        dht[KEY_TRACKER_STATUS] = "Working";
+    else
+        dht[KEY_TRACKER_STATUS] = "Disabled";
+
+    if (BitTorrent::Session::instance()->isPeXEnabled() && !torrent->isPrivate())
+        pex[KEY_TRACKER_STATUS] = "Working";
+    else
+        pex[KEY_TRACKER_STATUS] = "Disabled";
+
+    if (BitTorrent::Session::instance()->isLSDEnabled() && !torrent->isPrivate())
+        lsd[KEY_TRACKER_STATUS] = "Working";
+    else
+        lsd[KEY_TRACKER_STATUS] = "Disabled";
+
+    if (torrent->isPrivate()) {
+        dht[KEY_TRACKER_MSG] = privateMsg;
+        pex[KEY_TRACKER_MSG] = privateMsg;
+        lsd[KEY_TRACKER_MSG] = privateMsg;
+    }
+
+    uint seedsDHT = 0, seedsPeX = 0, seedsLSD = 0, peersDHT = 0, peersPeX = 0, peersLSD = 0;
+    foreach (const BitTorrent::PeerInfo &peer, torrent->peers()) {
+        if (peer.isConnecting()) continue;
+
+        if (peer.fromDHT()) {
+            if (peer.isSeed())
+                ++seedsDHT;
+            else
+                ++peersDHT;
+        }
+        if (peer.fromPeX()) {
+            if (peer.isSeed())
+                ++seedsPeX;
+            else
+                ++peersPeX;
+        }
+        if (peer.fromLSD()) {
+            if (peer.isSeed())
+                ++seedsLSD;
+            else
+                ++peersLSD;
+        }
+    }
+
+    dht[KEY_TRACKER_SEEDS] = seedsDHT;
+    dht[KEY_TRACKER_PEERS] = peersDHT;
+    pex[KEY_TRACKER_SEEDS] = seedsPeX;
+    pex[KEY_TRACKER_PEERS] = peersPeX;
+    lsd[KEY_TRACKER_SEEDS] = seedsLSD;
+    lsd[KEY_TRACKER_PEERS] = peersLSD;
+
+    stickyTrackers << dht;
```

Comment:
move `stickyTrackers` declaration above this line and initialize it, seems you're following C rules in modern C++ code... :\